Updated Upstream (Bukkit/CraftBukkit/Spigot)

Machine-Maker opened the pull request:
Updated Upstream (Bukkit/CraftBukkit/Spigot)

Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
7b1856bf PR-278: Add API to check and set if horses are grazing hay
0111185d PR-275: Add isJohnny / setJohnny for Vindicator
ea81107b PR-274: Add trap related methods to SkeletonHorse
eeda7998 Make EnderDragon extend/implement Mob
96509e04 PR-702: fromNodeTree is now private, this does not belong to the API
ead93bd8 PR-701: Option to configure the max width of a line in the config
becb47c4 PR-699: Add missing information to javadoc of ItemFlag#HIDE_POTION_EFFECTS
f3a8a4ce SPIGOT-5766: Updates related to the villager trading changes introduced in MC 1.14.
79cb3d7f PR-691: Add Material#getDefaultAttributeModifiers
65ff9376 SPIGOT-6866: Fixed spaces a between comment indicator and actual comment
8a747588 Restore protected members to YamlConfiguration
ca67e0f3 PR-693: Add proper javadocs for dolphins grace, bad omen, and hero of the village
3e2dd2bc SPIGOT-3247: Comment support for YAML files

CraftBukkit Changes:
c93fcba4 PR-378: Add API to check and set if horses are grazing hay
752fa511 PR-374: Add isJohnny / setJohnny for Vindicator
19f2af06 PR-373: Add trap related methods to SkeletonHorse
b8a25cbd Fix map colour not being marked as a handled tag
eea8d559 Make EnderDragon extend/implement Mob
633ac8b3 Add self-riding check to Entity#addPassenger
0ba6b901 SPIGOT-5766: The VillagerReplenishTradeEvent is now called when a villager restocks its trades.
bdff7d56 SPIGOT-5627: Leads vanish when transporting leashed entities through a Nether portal
319ca703 PR-978: Add Material#getDefaultAttributeModifiers
c59b0ca1 SPIGOT-6859: PlayerInteractEntityEvent#setCancelled desync with Axolotls

Spigot Changes:
78a14d9d SPIGOT-6644: Previously inactive experience orbs may not target players

kennytv reviewed (dismissed):
Spigot broke some config stuff, we'll either have to wait or fix them ourselves if there's more important commits to upstream

jpenilla commented on patches/api/0352-Add-system-prop-for-disabling-config-comment-parsing.patch:
```diff
@@ -0,0 +1,49 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 30 Dec 2021 16:35:56 -0800
+Subject: [PATCH] Add system prop for disabling config comment parsing
```
```suggestion
Subject: [PATCH] Add system prop for disabling config comment parsing by default

Allows for certain legacy plugins to continue to work without changes by setting `Paper.parseYamlCommentsByDefault` to false
```

Machine-Maker replied:
Done, didnt do exactly that commit msg cause I don't like when the patch file cuts off abruptly, but I think it gets the msg across

Machine-Maker committed:
Updated Upstream (Bukkit/CraftBukkit/Spigot)

Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
7b1856bf PR-278: Add API to check and set if horses are grazing hay
0111185d PR-275: Add isJohnny / setJohnny for Vindicator
ea81107b PR-274: Add trap related methods to SkeletonHorse
eeda7998 Make EnderDragon extend/implement Mob
96509e04 PR-702: fromNodeTree is now private, this does not belong to the API
ead93bd8 PR-701: Option to configure the max width of a line in the config
becb47c4 PR-699: Add missing information to javadoc of ItemFlag#HIDE_POTION_EFFECTS
f3a8a4ce SPIGOT-5766: Updates related to the villager trading changes introduced in MC 1.14.
79cb3d7f PR-691: Add Material#getDefaultAttributeModifiers
65ff9376 SPIGOT-6866: Fixed spaces a between comment indicator and actual comment
8a747588 Restore protected members to YamlConfiguration
ca67e0f3 PR-693: Add proper javadocs for dolphins grace, bad omen, and hero of the village
3e2dd2bc SPIGOT-3247: Comment support for YAML files

CraftBukkit Changes:
c93fcba4 PR-378: Add API to check and set if horses are grazing hay
752fa511 PR-374: Add isJohnny / setJohnny for Vindicator
19f2af06 PR-373: Add trap related methods to SkeletonHorse
b8a25cbd Fix map colour not being marked as a handled tag
eea8d559 Make EnderDragon extend/implement Mob
633ac8b3 Add self-riding check to Entity#addPassenger
0ba6b901 SPIGOT-5766: The VillagerReplenishTradeEvent is now called when a villager restocks its trades.
bdff7d56 SPIGOT-5627: Leads vanish when transporting leashed entities through a Nether portal
319ca703 PR-978: Add Material#getDefaultAttributeModifiers
c59b0ca1 SPIGOT-6859: PlayerInteractEntityEvent#setCancelled desync with Axolotls

Spigot Changes:
78a14d9d SPIGOT-6644: Previously inactive experience orbs may not target players
